controllers/machinehealthcheck_controller.go

```diff
@@ -376,54 +373,21 @@ func (r *MachineHealthCheckReconciler) getMachineFromNode(nodeName string) (*clu
 	return &machineList.Items[0], nil
 }
 
-func (r *MachineHealthCheckReconciler) watchClusterNodes(ctx context.Context, c client.Client, cluster *clusterv1.Cluster) error {
-	key := util.ObjectKey(cluster)
-	if _, ok := r.getClusterCache(key); ok {
-		// watch was already set up for this cluster
+func (r *MachineHealthCheckReconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	// If there is no tracker, don't watch remote nodes
+	if r.Tracker == nil {
 		return nil
 	}
```

Comment:
Is this a valid use case we want to support?

Reply:
Probably not, I had added it so that a tracker wouldn't necessarily be needed for testing, I can remove it though if you think we should